### ISSUE TITLE
[documentation] sdk documentation of kfp.v2

### ISSUE THREAD

**Issue by verhoek** [kubeflow/pipelines#8201]
I am having difficulties regarding the documentation of the python SDK of pipelines, in particular the kfp.v2 package of the 1.8.x releases.
Some functionality from the package kfp.v2.google seems to be referenced on the web/from the vertex ai docu, yet I cannot find any official referenced documentation of it from the project itself, for example on https://kubeflow-pipelines.readthedocs.io .
I understand by browsing through the issues that apparently kfp.v2.google contains deprecated code such as the AIPlatformClient.

The effect is that for older versions of kfp there is no SDK documentation available of all that was once/is in kfp.v2 release 1.8.x 
that is later actually removed in kfp release 2.x . I think this is very confusing for people new to kfp, for which the stable release is still 1.8.x, but from tutorials from vertex ai use code from kfp.v2. I would expect this documentation, even now deprecated, is still available in the SDK documentation.

PS: Also, what was the original intent of the v2 package? Already making available v2 sdk functionality from 1.8.x ? 
In the 2.x versions the kfp.v2 seems to point back to the main package kfp itself, which probably makes sense.

**Comment by connor-mccarthy** [comment]
@verhoek, thanks for raising this. We are working on a significant documentation refresh that should help with this. In the meantime, maybe I can help here.

Which version of the SDK are you using?

> Some functionality from the package kfp.v2.google seems to be referenced on the web/from the vertex ai docu

Can you point me in the direction of this? It will be easier to speak in specifics based on the functionality you're trying to use from the Vertex docs.

> PS: Also, what was the original intent of the v2 package? Already making available v2 sdk functionality from 1.8.x ?

The v2 subpackage/namespace within 1.8.x was intended to 1) provide the some of the KFP SDK v2 authoring experience within the 1.8.x and 2) enable compilation to the new IR pipeline definition protocol. Both should make for an easier migration to 2.0.0.

> In the 2.x versions the kfp.v2 seems to point back to the main package kfp itself, which probably makes sense.

Yes, this is for backward compatibility of existing user import code.

>  would expect this documentation, even now deprecated, is still available in the SDK documentation.

Reference documentation on Read The Docs has not historically been pinned for all released versions, unfortunately. This is changing beginning with the 2.x.x stable release (and some 2.x.x beta releases). Reference docs are generated based on docstrings, so that is one way to find historical reference docs. Please let me know if I'm missing something about this comment.

**Comment by verhoek** [comment]
> 
> Which version of the SDK are you using?
> 

It was 1.8.13. 

> > Some functionality from the package kfp.v2.google seems to be referenced on the web/from the vertex ai docu
> 
> Can you point me in the direction of this? It will be easier to speak in specifics based on the functionality you're trying to use from the Vertex docs.

We tried to use the deprecated package kfp.v2.google for the aiplatformclient, which seems to have been moved to the vertex ai sdk in the google.cloud package. When I check the vertex ai documentation though, a hint has been added that one should use the google.cloud package. I don't know if that was there before or we didn't check properly - I retract the statement that vertex ai docu references to the kfp.v2.google package, though it does refer to the kfp.v2 

We tried to submit/schedule a job and came across https://cloud.google.com/vertex-ai/docs/pipelines/schedule-cloud-scheduler where kfp.v2 is used but undocumented for 1.8.x .

> 
> > PS: Also, what was the original intent of the v2 package? Already making available v2 sdk functionality from 1.8.x ?
> 
> The v2 subpackage/namespace within 1.8.x was intended to 1) provide the some of the KFP SDK v2 authoring experience within the 1.8.x and 2) enable compilation to the new IR pipeline definition protocol. Both should make for an easier migration to 2.0.0.
> 
> > In the 2.x versions the kfp.v2 seems to point back to the main package kfp itself, which probably makes sense.
> 
> Yes, this is for backward compatibility of existing user import code.
> 
> > would expect this documentation, even now deprecated, is still available in the SDK documentation.
> 
> Reference documentation on Read The Docs has not historically been pinned for all released versions, unfortunately. This is changing beginning with the 2.x.x stable release (and some 2.x.x beta releases). Reference docs are generated based on docstrings, so that is one way to find historical reference docs. Please let me know if I'm missing something about this comment.

Okay, sounds reasonable, thanks for clarification!

**Comment by connor-mccarthy** [comment]
> We tried to use the deprecated package kfp.v2.google for the aiplatformclient, which seems to have been moved to the vertex ai sdk in the google.cloud package.

That is correct. For submitting jobs to Vertex, use `google.cloud.aiplatform.PipelineJob(...).submit()` from the Vertex SDK.

> We tried to submit/schedule a job and came across https://cloud.google.com/vertex-ai/docs/pipelines/schedule-cloud-scheduler where kfp.v2 is used but undocumented for 1.8.x .

I see. This may already be clear, but just to be sure: there is a v2 namespace in `kfp==1.8.x`. There is also `kfp==2.x.x` which is currently in beta stage. The docs you are referring to are about the v2 namespace in v1, which applies to `kfp==1.8.x`.

The v2 namespace in `kfp==1.8.x` will compile to IR YAML, a pipeline protocol format that Vertex AI can read and execute. The main namespace in `kfp==1.8.x` will compile to Argo workflow, which Vertex cannot read and execute. If you wish to use Vertex as your BE, use the v2 namespace.

**Comment by connor-mccarthy** [comment]
@verhoek, the following docs will likely be helpful for you: https://www.kubeflow.org/docs/components/pipelines/v2/migration/
